papr/testrunner

```diff
@@ -582,6 +582,14 @@ s3_upload() {
         fi
     fi
 
+    # go through every file we'll upload and make sure it's no more than the max
+    # size, otherwise violently truncate it
+    find $upload_dir -mindepth 1 -size +3M | while read f; do
+        local orig_size=$(stat -c %s "$f")
+        truncate -s 3M "$f"
```

Comment:
We generally don't want to truncate, we want to...um `tail` it right?  I am offhand not sure if there's an easy way to do that short of clone/copy to new file then rename.

Reply:
I debated a bit between `tail` and (what is essentially) `head`. If we assume going over 3M means something went wrong, then ISTM logs prior to things going haywire are more important than the tail, right? (Which could be the same non-sensical thing over and over).